lib/money/bank/open_exchange_rates_bank.rb

```diff
@@ -174,6 +174,7 @@ def source
       #
       # @return [Array] Array of exchange rates
       def update_rates
+        store.each_rate { |iso_from, iso_to| add_rate(iso_from, iso_to, nil) }
```

Comment:
I think we should do it inside transaction, so no other thread hit in situation, when old cache reseted, and new rate not added yet.

```suggestion
        store.transaction do
          store.each_rate { |iso_from, iso_to| add_rate(iso_from, iso_to, nil) }
          exchange_rates.each do |exchange_rate|
            rate = exchange_rate.last
            currency = exchange_rate.first
            next unless Money::Currency.find(currency)
            set_rate(source, currency, rate)
            set_rate(currency, source, 1.0 / rate)
          end
        end
```

Reply:
I think the better way to clear only calculated rates after new `exchange_rates` setup, to not block hash for another threads. what do you think?

Reply:
Other threads in this case will try to `update_rates` too, and will be blocked in transaction anyway: on first `set_rate` call, and then they wait each other.

> clear only calculated rates after new exchange_rates setup

Moreover, without explicit transaction call, Thread#1 can clear good fresh rate, setted by Thread#2 moment ago.

Reply:
I agree for the transaction, since this has been added on 6.13 and other part of the code base are not updated I will work on this later

Reply:
Hi @spk, here is transaction fix #66 😉

Reply:
hi @korun thanks for the fix <3